File: src/main/kotlin/javaset/model/TestBoxMode.java
package javaset.model;

import java.util.Objects;

public class TestBoxMode {

    int id;
    String name;

    public TestBoxMode(int id, String name) {
        this.id = id;
        this.name = name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof TestBoxMode)) return false;
        TestBoxMode that = (TestBoxMode) o;
        return id == that.id && Objects.equals(name, that.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name);
    }

    @Override
    public String toString() {
        return "TestBoxMode{" +
                "id=" + id +
                ", name='" + name + '\'' +
                '}';
    }
}

File: src/main/kotlin/kotlinset/model/TestBoxMode.kt
package kotlinset.model

data class TestBoxMode(
    val id: Int,
    val name: String
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is TestBoxMode) return false

        if (id != other.id) return false
        if (name != other.name) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id
        result = 31 * result + name.hashCode()
        return result
    }

    override fun toString(): String {
        return "TestBoxMode(id=$id, name='$name')"
    }
}
